src/components/UserProfile.jsx

```diff
@@ -0,0 +1,413 @@
+    setEditMode(!editMode);
+  };
+
+  const handleCancelEdit = () => {
+    fetchMyUserData();
+    fetchUserData();
+    setEditMode(false);
+  };
+
+  const handleSaveProfile = async () => {
+    if (!userNameAux) {
+      setConfirmationMessage("Name is required");
+      setShowConfirmation(true);
+      return;
+    }
+    try {
+      const response = await fetch(`${'http://127.0.0.1:8000'}/user/${userId}`, {
+        method: 'PUT',
+        headers: {
+          'Authorization': `Bearer ${token}`,
+          'Content-Type': 'application/json',
+        },
+        body: JSON.stringify({
+          username: userNameAux,
+          email: userMailAux,
+          bio: userBioAux,
+        }),
+      });
+  
+      if (!response.ok) {
+        throw new Error('Failed to update user data');
+      }
+  
+      const updatedData = await response.json();
+  
+      setUserName(updatedData.username);
+      setUserMail(updatedData.email);
+      setUserBio(updatedData.bio || '');
+  
+      setConfirmationMessage("Profile updated successfully");
+    } catch (error) {
+      console.error('Error updating user data:', error);
+      setConfirmationMessage("Failed to update profile");
+    }
+  
+    setShowConfirmation(true);
+    setEditMode(false);
+  };
+
+  const handleImageUpload = async () => {
+    const userData = {
+    };
+    const fileInput = document.createElement('input');
+    fileInput.type = 'file';
+    fileInput.accept = 'image/*';
+    fileInput.onchange = async (e) => {
+      const file = e.target.files[0];
+      if (file && file.type.startsWith('image/')) {
+        const imageUrl = URL.createObjectURL(file);
+        setProfilePicture(imageUrl);
+  
+        const formData = new FormData();
+        formData.append("file", file);
+  
+        try {
+          const response = await fetch(`http://127.0.0.1:8000/user/${userId}`, {
+            method: 'PUT',
+            headers: {
+              'Authorization': `Bearer ${token}`,
+            },
+            body: formData,
+          });
+  
+          if (!response.ok) {
+            throw new Error('Failed to update user data');
+          }
+  
+          const data = await response.json();
+          setConfirmationMessage("Profile updated successfully");
+        } catch (error) {
+          console.error('Error updating user data:', error);
+          setConfirmationMessage("Failed to update profile");
+        }
+        setShowConfirmation(true);
+      }
+    };
+    fileInput.click();
+  };
+
+  const handleImageRemove = (event) => {
+    setRemoveQuestion(false)
+    setProfilePicture("")
+  };
+
+  const handleImageRemoveQuestion = (event) => {
+    setRemoveQuestion(true);
+  };
+
+  const ImageRemoveQuestionClose = (event) => {
+    setRemoveQuestion(false)
+  };
+
+  const handleConfirmationClose = () => {
+    setShowConfirmation(false);
+  };
+
+  const getRecipes = () => {
+    fetch(`http://127.0.0.1:8000/recipe/user/${userName}`, {
```

Comment:
You should use the enironment variable with the backend address to also work when deployed

Reply:
I know it, when I mark as complete the US I will use the env variable.

Reply:
👍